### DISCUSSION TITLE
Should Microsoft casks really depend on `microsoft-auto-update`?

### DISCUSSION THREAD

**Discussion by tadamcz** [Homebrew/discussions#5481] [Casks] [answered]
### Output of `brew config`

```shell
NA
```

### Output of `brew doctor`

```shell
NA
```

### Description of issue

All Microsoft casks depend on `microsoft-auto-update`, as you can see here: https://github.com/search?q=org%3AHomebrew++depends_on+cask%3A+%22microsoft-auto-update%22&type=code

This sort of thing seems odd to me -- isn't the whole point of Homebrew to handle applications and their updates centrally?

Is there a particular reason it's been done this way?

(btw, `microsoft-auto-update` requires sudo every time you upgrade it with `brew upgrade`, and presumably also when you install it)

### Relevant casks

microsoft-teams

**Comment by SMillerDev** [comment]
> This sort of thing seems odd to me -- isn't the whole point of Homebrew to handle applications and their updates centrally?

Yes, but Microsoft doesn't optimise their software for Homebrew installs so it requires their updater.

**Reply by tadamcz** [reply]
So, the main application won't work at all without the updater? What would happen if you installed and ran it?

**Reply by SMillerDev** [reply]
I'm not sure, you can try uninstalling it. But I'm pretty sure the dependency would be omitted if everything worked fine without it.

**Comment by cbean** [comment]
Tested for years now, the package microsoft-auto-update can be uninstalled afterwards. Just use:

`brew uninstall microsoft-auto-update`

**Comment by rkourdis** [comment]
Tangentially related; how does `brew` knows to use `sudo` when installing `microsoft-auto-update` or,  for example, `microsoft-teams`? I'm looking at the [formulae](https://github.com/Homebrew/homebrew-cask/blob/8236b78a6ee0f1f95e8d81b68a570acd83a85f14/Casks/m/microsoft-auto-update.rb) for both and there's nothing about the installer requiring root.

EDIT: Looking at [this](https://github.com/Homebrew/brew/blob/77b3a6e8486d2ac71bc7adb973ad74b835da1c80/Library/Homebrew/cask/artifact/pkg.rb#L64) code it seems that all `.pkg` installs need `sudo`. That would make sense, if correct. Apologies for the naive question, I've only recently started using Mac.

**Reply by p-linnane** [reply]
You're correct. `.pkg` installers use `sudo`.

**Comment by gh10ltz** [comment]
is there any way at all we can break the dependency here apart from immediately brew uninstall microsoft-auto-update after an office package upgrade? this is so frustrating and intrusive

**Reply by gh10ltz** [reply]
@p-linnane ?

**Reply by p-linnane** [reply]
You can open a PR to try and remove the dependency.

**Reply by gh10ltz** [reply]
Doesn't seem like I can open a PR for this. @EricFromCanada is this something you could please review?

**Reply by gh10ltz** [reply]
@krehel would you be able to get this change added?

**Reply by SMillerDev** [reply]
Please don't tag random maintainers. Anyone can make pull requests, and it's the only way to request changes in Homebrew

**Reply by gh10ltz** [reply]
These aren't random maintainers. These are the individuals who are typically handling cask updates.

**Comment by elgs** [comment]
I run brewall everyday.
```
alias brewdeps="brew deps --installed --tree"
alias brewup="brew upgrade --greedy"
alias caskup="brew upgrade --cask --greedy"
alias brewcl="brew cleanup -s && rm -rf \$(brew --cache)"
alias brewrm-msupdate="brew uninstall microsoft-auto-update"
alias brewall="brewup && caskup && brewcl && brewrm-msupdate"
```

**Reply by bevanjkay** [reply]
Feel free to open a PR to remove the dependency from the `microsoft-*` casks - it likely isn't required.

**Reply by gh10ltz** [reply]
Yep... I run something similar:
```shell
  _do_cleanup() {
    _run env HOMEBREW_NO_AUTO_UPDATE=1 brew cleanup --prune=all || return 1
    if _run env HOMEBREW_NO_AUTO_UPDATE=1 brew list --cask microsoft-auto-update &>/dev/null; then
      _run env HOMEBREW_NO_AUTO_UPDATE=1 brew uninstall microsoft-auto-update
      echo "Microsoft AutoUpdate uninstalled, proceeding with related cleanup..."
      sudo sh -c '
        rm -f /Library/LaunchDaemons/com.microsoft.office.*.plist;
        chflags nouchg /Library/LaunchDaemons /Library/LaunchAgents;
        chflags uchg /Library/LaunchDaemons /Library/LaunchAgents;
        def="/Library/Application Support/Microsoft/Defender/uninstall/uninstall";
        [ -x "$def" ] && "$def";
        rm -rf "/Applications/Microsoft Defender.app";
      '
    fi
  }
```

So wasteful and useless. At no point in time was this garbage ever a necessary dep and there is no reason why it should be still. How more people aren't complaining about this (especially given the LaunchAgent and LaunchDaemon plists installed) is beyond me...

**Reply by bevanjkay** [reply]
Again, you're more than welcome to open a PR to make the changes.

**Reply by gh10ltz** [reply]
Thanks Bevan. Sadly, my GH account is considered "unverified" and as a result I'm quite limited in what I am able to do on it. Perhaps someone else can squeeze this through.

**Comment by christopherowen** [comment, marked as answer]
Follow-up PR: Homebrew/homebrew-cask#235372
